src/index_map.rs

```diff
@@ -235,3 +235,12 @@ impl<K: Eq + Hash, V, S: Default> Default for FrozenIndexMap<K, V, S> {
         }
     }
 }
+
+impl<K: Clone, V: Clone, S: Clone> Clone for FrozenIndexMap<K, V, S> {
+    fn clone(&self) -> Self {
+        Self {
+            map: unsafe { self.map.get().as_ref().unwrap() }.clone().into(),
+            in_use: self.in_use.clone(),
```

Comment:
issue: no, we should assert that `in_use` is false and then temporaily set it to true before cloning all elements, after which we set it back to false (and the new object should have it be false too)

Reply:
Fixed in [`64154a3` (#57)](https://github.com/Manishearth/elsa/pull/57/commits/64154a3a1c12017b5b8bcce299c998dce8f9ded2)